### PR TITLE
Add .editorconfig file

### PR DESCRIPTION
**Pull request by newbazz**
It specifies the indentation of
a file irrespective of the text
editor or IDE.

Closes https://github.com/coala/coala/issues/4097

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by newbazz** [comment]
@jayvdb sir i have reopened pr from same branch then y a new pr?

**Comment by newbazz** [comment]
please review.

**Comment by newbazz** [comment]
@jayvdb sorry for the  inconvenience.

**Comment by prashant0598** [comment]
@newbazz from next time please try not to open so many pr for an issue.
Keep Contributing in coala :smile:

**Comment by prashant0598** [comment]
ack e687578

**Comment by newbazz** [comment]
@prashant0598 okk sure sorry and thanks. :P

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,19 @@
+root = true
+
+[**]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+max_line_length = 80
+
+# Set default charset and indentation for python files
+[{*.py,coalib/**/*.py,./coala,tests/**/*.py}]
```
Comment (aashutoshrathi) [inline .editorconfig]: @newbazz add spaces after `'` in line 10 and 16.

**Comment by aashutoshrathi** [comment]
Also, tick the checkpoints that are appropriate in PR template.

File: .editorconfig (hunk 1 above)
Comment (newbazz) [reply .editorconfig]: no spaces to be added after `,` was discussed earlier

Comment (aashutoshrathi) [reply .editorconfig]: Oh my bad.

**Comment by newbazz** [comment]
@aashutoshrathi i should wait right?

**Comment by Makman2** [comment]
@gitmate-bot rebase

**Comment by Makman2** [comment]
@gitmate-bot rebase

**Comment by Makman2** [comment]
@rultor merge

**Pull request merged**